inst/perl/t/03-oneshot_analysis.t

```diff
@@ -0,0 +1,518 @@
+  },
+  bd => {
+    params => {name => 'blognorb', cellsizes => [100,100]},
+    data   => $gp_lb,
+  },
+  analysis_config => {
+    calculations => ['calc_endemism_central'],
+  },
+};
+
+my $t_msg_suffix = 'default config, raster files';
+$t->post_ok('/analysis_spatial_oneshot' => json => $oneshot_data)
+    ->status_is(200, "status, $t_msg_suffix");
+}
+
+my %common_args = (
+    bd => {
+        params       => { name => 'blognorb', cellsizes => [ 500, 500 ] },
+        data        => $gp_lb,
+    },
+    analysis_config => {
+        calculations => [qw /calc_endemism_central calc_pd calc_redundancy/],
+    },
+    tree => $tree,
+);
+
+{
+my $oneshot_data = {
+    shapefile_params => {
+            files => [ "$data_dir/r1.shp", "$data_dir/r2.shp", "$data_dir/r3.shp" ],
+            group_field_names      => [ qw/:shape_x :shape_y/ ],
+            label_field_names      => [ 'label' ],
+            sample_count_col_names => [ 'count' ]
+    },
+    %common_args,
+};
+
+my $t_msg_suffix = 'default config, raster files';
+$t->post_ok('/analysis_spatial_oneshot' => json => $oneshot_data)
+    ->status_is(200, "status, $t_msg_suffix");
+}
+
+{
+my $oneshot_data = {
+    delimited_text_params => {
+            files => [ "$data_dir/r1.csv", "$data_dir/r2.csv", "$data_dir/r3.csv" ],
+            group_columns        => [ 1, 2 ],
+            label_columns        => [ 4 ],
+            sample_count_columns => [ 3 ],
+        },
+    %common_args,
+};
+
+my $t_msg_suffix = 'default config, raster files';
+$t->post_ok('/analysis_spatial_oneshot' => json => $oneshot_data)
+    ->status_is(200, "status, $t_msg_suffix");
+}
+
+{
+my $oneshot_data = {
+    spreadsheet_params => {
+            files                  => [ "$data_dir/r1.xlsx", "$data_dir/r2.xlsx", "$data_dir/r3.xlsx" ],
+            group_field_names      => [ qw/X Y/ ],
+            label_field_names      => [ 'label' ],
+            sample_count_col_names => [ 'count' ]
+        },
+    %common_args,
+};
+
+my $t_msg_suffix = 'default config, raster files';
+$t->post_ok('/analysis_spatial_oneshot' => json => $oneshot_data)
+    ->status_is(200, "status, $t_msg_suffix");
+}
+
+my %common_args = (
+    bd => {
+        params       => { name => 'blognorb', cellsizes => [ 500, 500 ] },
+    },
+    analysis_config => {
+        calculations => [qw /calc_endemism_central calc_pd calc_redundancy/],
+    },
+    tree => $tree,
+);
+
+{
+my $oneshot_data = {
+    raster_params => {
+            files => [ "$data_dir/r1.tif", "$data_dir/r2.tif", "$data_dir/r3.tif" ]
+        },
+    spreadsheet_params => {
+            files                  => [ "$data_dir/r1.xlsx", "$data_dir/r2.xlsx", "$data_dir/r3.xlsx" ],
+            group_field_names      => [ qw/X Y/ ],
+            label_field_names      => [ 'label' ],
+            sample_count_col_names => [ 'count' ]
+        },
+    %common_args,
+};
+
+my $t_msg_suffix = 'default config, raster files';
```

Comment:
Need to update the test name to include spreadsheets so debugging is easier.  Same for later cases.

Reply:
Resolved by recent commits